packages/gsul/src/getStatusUpdates.ts

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Logger} from "@aws-lambda-powertools/logger"
+import {injectLambdaContext} from "@aws-lambda-powertools/logger/middleware"
+import {DynamoDBClient} from "@aws-sdk/client-dynamodb"
+import {DynamoDBDocumentClient, QueryCommandInput} from "@aws-sdk/lib-dynamodb"
+import middy from "@middy/core"
+import inputOutputLogger from "@middy/input-output-logger"
+import validator from "@middy/validator"
+import {transpileSchema} from "@middy/validator/transpile"
+import {errorHandler} from "./errorHandler.ts"
+import {runDynamoDBQueries} from "./dynamoDBclient.ts"
+import {requestSchema, requestType, inputPrescriptionType} from "./schema/request.ts"
+import {responseType, outputPrescriptionType, itemType} from "./schema/response.ts"
+import {DynamoDBResult} from "./schema/result.ts"
+
+const logger = new Logger({serviceName: "GSUL"})
+const client = new DynamoDBClient({region: "eu-west-2"})
+const docClient = DynamoDBDocumentClient.from(client)
+const tableName = process.env.TABLE_NAME
+
+const lambdaHandler = async (event: requestType): Promise<responseType> => {
+  // there are deliberately no try..catch blocks in this as any errors are caught by custom middy error handler
+  // and an error response is sent
+
+  const queryParams = event.prescriptions.map((prescription) => {
+    // create query for each prescription and ods code passed in
+    const queryParam : QueryCommandInput = {
+      TableName: tableName,
+      IndexName: "PharmacyODSCodePrescriptionIDIndex",
+      KeyConditionExpression: "PrescriptionID = :inputPrescriptionID AND PharmacyODSCode = :inputPharmacyODSCode",
+      ExpressionAttributeValues: {
+        ":inputPharmacyODSCode": prescription.odsCode,
+        ":inputPrescriptionID": prescription.prescriptionID
+      }
+    }
+    return queryParam
+  })
+
+  // run the dynamodb queries
+  const queryResultsTasks = runDynamoDBQueries(queryParams, docClient, logger)
+
+  // get all the query results
+  const queryResults = await Promise.all(queryResultsTasks)
+
+  // queryResults is an array of arrays
+  // the first array is for each prescription id passed in as we have created a query for each prescription
+  // and the second array is the items for that prescription id
+  // we use flat() to just get an array of items which we can pass to buildResults
+  const itemResults = buildResults(event.prescriptions, queryResults.flat())
```

Comment:
we could build the results one prescription at a time from the results of each query and very much simplify the code below.